### PR TITLE
Reinclude symbol parsing

### PR DESCRIPTION
**Pull request by naomijub**

### CODE REVIEW

**Comment by naomijub** [comment]
from issue https://github.com/naomijub/edn-rs/issues/59

File: src/deserialize/parse.rs [hunk 1]
```diff
@@ -54,6 +52,23 @@ fn read_str(chars: &mut std::str::Chars) -> Edn {
     Edn::Str(string)
 }
 
+fn read_others(a: char, chars: &mut std::str::Chars) -> Result<Edn, Error> {
```
Comment (evaporei) [inline src/deserialize/parse.rs]: I don't know if I like the `others` suffix :thinking:

File: src/deserialize/parse.rs [hunk 2]
```diff
@@ -54,6 +52,23 @@ fn read_str(chars: &mut std::str::Chars) -> Edn {
...
+    let c_len = chars
+        .clone()
+        .enumerate()
+        .take_while(|&(i, c)| i <= 200 && !c.is_whitespace() && c != ')' && c != '}' && c != ']')
```
Comment (evaporei) [inline src/deserialize/parse.rs:59]: Why the 200 here?

File: src/deserialize/parse.rs (hunk 1 above)
Comment (naomijub) [reply src/deserialize/parse.rs]: c49169c

File: src/deserialize/parse.rs (hunk 2 above)
Comment (naomijub) [reply src/deserialize/parse.rs:59]: it is like a `timeout`

File: src/deserialize/parse.rs [hunk 3]
```diff
@@ -54,6 +52,23 @@ fn read_str(chars: &mut std::str::Chars) -> Edn {
...
+        .count();
+    match a {
+        ws if ws.is_whitespace() => Err(Error::ParseEdn(format!("\"{}\" could not be parsed", ws))),
+        c => {
+            let mut symbol = String::from(c);
+            let symbol_chars = chars.take(c_len).collect::<String>();
+            symbol.push_str(&symbol_chars);
+            Ok(Edn::Symbol(symbol))
+        }
+    }
```
Comment (evaporei) [inline src/deserialize/parse.rs]: (style suggestion)
Do you think we will have more cases?
What do you think of 

```rust
if a.is_whitespace() {
    return Err(Error::ParseEdn(format!("\"{}\" could not be parsed", ws)));
}
let mut symbol = String::from(c);
let symbol_chars = chars.take(c_len).collect::<String>();
symbol.push_str(&symbol_chars);
Ok(Edn::Symbol(symbol))
```

Comment (evaporei) [reply src/deserialize/parse.rs]: Actually `c` and `c_len` variables don't exist on my example.

Comment (naomijub) [reply src/deserialize/parse.rs]: this was my first idea, but then we changed to `read_symbol` and it makes no more sense to have a match...

Comment (naomijub) [reply src/deserialize/parse.rs]: 92520e8

File: src/deserialize/parse.rs (hunk 2 above)
Comment (naomijub) [reply src/deserialize/parse.rs:59]: 92520e8

**Review by evaporei — APPROVED** [review]
:shipit:

**Pull request merged**